llpc/patch/llpcPatchBufferOp.cpp

```diff
@@ -1521,31 +1520,62 @@ bool PatchBufferOp::RemoveUsersForInvariantStarts(
 }
 
 // =====================================================================================================================
-// Replace a fat pointer load with the intrinsics required to do the load.
-Value* PatchBufferOp::ReplaceLoad(
-    LoadInst* const pLoadInst) // [in] The load instruction to replace.
+// Replace a fat pointer load or store with the required intrinsics.
+Value* PatchBufferOp::ReplaceLoadStore(
+    Instruction* const pInst) // [in] The instruction to replace.
 {
-    m_pBuilder->SetInsertPoint(pLoadInst);
+    LoadInst* const pLoadInst = dyn_cast<LoadInst>(pInst);
```

Comment:
pInst may be null. You could take it as a reference and not worry at all, or assert at the very beginning, or use dyn_cast_or_null